### ISSUE TITLE
412 Precondition Failed when trying to install from a chef server to a node (Vagrant)

### ISSUE THREAD

**Issue by rricard** [RiotGamesCookbooks/rbenv-cookbook#30]
Hi,

I'm fairly new to chef, so it's not impossible that I did something wrong but when I'm trying to provision a node, I end up with this error, which only seems to happen with this cookbook :

```
HTTP Request Returned 412 Precondition Failed: messageUnable to solve constraints, the following solutions were attempted 

    Unable to satisfy goal constraint rbenv due to constraints on ohai, runit
        (rbenv = 1.4.1) -> (ohai >= 1.1.0)
        (rbenv = 1.4.1) -> (git = 2.5.0) -> (runit >= 1.0.0)
        (rbenv = 1.4.1) -> (git = 2.5.0) -> (build-essential = 1.4.0)
        (rbenv = 1.4.1) -> (git = 2.5.0) -> (dmg = 1.1.0)
        (rbenv = 1.4.1) -> (git = 2.5.0) -> (windows = 1.8.10) -> (chef_handler = 1.1.4)
        (rbenv = 1.4.1) -> (git = 2.5.0) -> (yum = 2.2.0)
most_constrained_cookbooks(runit >= 1.0.0)(ohai >= 1.1.0)non_existent_cookbooksunsatisfiable_run_list_itemrbenv

================================================================================
Error Resolving Cookbooks for Run List:
================================================================================

Missing Cookbooks:
------------------


Expanded Run List:
------------------
* nginx
* rbenv
```

I think there's something broken with the dependencies (I use berkshelf to manage my cookbooks) but I can't figure out what ... I've tried to find more informations about this error but all I can find is the source code of chef on github that raise this error.

**Comment by MattMencel** [comment]
Did you figure this out?  We're have the exact same problem, though with different cookbooks.

Matt

**Comment by rricard** [comment]
Not yet. Didn't work on it since the problem occured.

**Comment by jasonwbarnett** [comment]
FYI- I found that these messages are extremely vague and rarely pinpoint the exact dependency resolution failure.

I was able to resolve my issue by narrowing down the node's run list. I found that a cookbook earlier in my run list had dependencies which conflicted with another cookbook in the run list. And even though there were only about 4 dependency conflicts, it listed about 25.

I hope this helps someone trying to figure this out for themselves.
